packages/flutter_tools/lib/src/fuchsia/fuchsia_device.dart

```diff
@@ -53,17 +53,20 @@ Future<vm_service.VmService> _kDefaultFuchsiaIsolateDiscoveryConnector(Uri uri)
   return connectToVmService(uri);
 }
 
-Future<DartDevelopmentService> _kDefaultFuchsiaIsolateDiscoveryDDSConnector(
+Future<Uri> _kDefaultFuchsiaIsolateDiscoveryDDSConnector(
+  FuchsiaDevice device,
   Uri remoteVmServiceUri,
   Uri serviceUri,
-  ) {
-  return DartDevelopmentService.startDartDevelopmentService(
+  ) async {
+  final Stream<Uri> dds = await device.dds.startDartDevelopmentService(
     remoteVmServiceUri,
-    serviceUri: serviceUri,
+    serviceUri.port,
     // TODO(bkonyi): enable authentication codes when we can enable
     // them on Fuchsia.
```

Comment:
Does this have a bug associated with it?

Reply:
Apparently not. It's just documented here: https://github.com/dart-lang/sdk/blob/3437336ed0004d6cc435e9ff04eb2425b0399f4f/sdk_nnbd/lib/_internal/vm/bin/vmservice_server.dart#L171

I'll file an issue.